packages/story-editor/src/app/media/utils/useMediaUploadQueue/constants.js

```diff
@@ -15,6 +15,8 @@
  */
 export const ITEM_STATUS = {
   PENDING: 'PENDING',
+  PREPARING: 'PREPARING',
+  PENDING_TRANSCODING: 'PENDING_TRANSCODING',
```

Comment:
Introduced these two new states to make the whole hook more maintainable and prevent items from being processed twice at a specific step (i.e. do not run mediainfo twice on the same video)

Best reviewed by looking at the effects in the hook and how items flow from pending -> preparing -> pending_transcoding -> transcoding -> uploading

Reply:
Update this comment. https://github.com/GoogleForCreators/web-stories-wp/blob/690006d36e37b0972af1a4b8875d8227db97d688/packages/story-editor/src/app/media/utils/useMediaUploadQueue/index.js#L64

Reply:
What about adding these new states to here

https://github.com/GoogleForCreators/web-stories-wp/blob/dba641ba6c13e3313ad3313fd0c165ca99ea6eaa/packages/story-editor/src/app/media/utils/useMediaUploadQueue/index.js#L733-L744

https://github.com/GoogleForCreators/web-stories-wp/blob/dba641ba6c13e3313ad3313fd0c165ca99ea6eaa/packages/story-editor/src/app/media/utils/useMediaUploadQueue/index.js#L644-L651

https://github.com/GoogleForCreators/web-stories-wp/blob/dba641ba6c13e3313ad3313fd0c165ca99ea6eaa/packages/story-editor/src/app/media/utils/useMediaUploadQueue/index.js#L588-L596